internal/service/resourcepolicy/resource.go

```diff
@@ -35,6 +38,39 @@ type resourcePolicyRS struct {
 	config.RSCommon
 }
 
+var charSetAlphaNum = []rune("abcdefghijklmnopqrstuvwxyz012346789")
+
+func randPolicyName(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = charSetAlphaNum[rand.Intn(len(charSetAlphaNum))] //nolint:gosec // This is not a security-sensitive operation, only to avoid POLICY_CANNOT_CONTAIN_A_DUPLICATE_NAME error
```

Comment:
seems strange that if we are calling a validate method, that the error POLICY_CANNOT_CONTAIN_A_DUPLICATE_NAME can appear

Reply:
Agree. Added a ticket: https://jira.mongodb.org/browse/CLOUDP-274990 to upstream

Reply:
i would add some "TEMPORARY" doc to this code until we API is fixed and we don't need random names

Reply:
Addressed in 5feb4e458bba8aed693dba4d5acd08c1c98823f8